### DISCUSSION TITLE
Wechselrichter Offline / Time Calibration Issue

### DISCUSSION THREAD

**Discussion by kerusker** [tbnobody/OpenDTU#1950] [Q&A] [answered]
In unregelmäßigen Abständen verliert die OpenDTU Kontakt zum Wechselrichter HM-1600. 

Im Ereignisprotokoll erscheint plötzliche dutzend Male die Meldung ID2 "Time Calibration" - jeweils mit Abstand von 5 Sekunden, anschließend bricht der Kontakt ab und die DTU geht offline. 

Der Wechselrichter produziert weiter Strom.
Der Fehler erscheint am Tag. 
Es wird nur ein Wechselrichter betrieben.
Unvermittelt bekommt die OpenDTU wieder Kontakt zum Wechselrichter. 

Folgende Lösungsversuche waren NICHT erfolgreich: 
Manuell "Time Calibration" 
Veränderung der Frequenz von 865MHz auf 868MHz
Austausch CMT Modul
Firmware Update
ESP Restet

![openDTU Ereignisanzeige](https://github.com/tbnobody/OpenDTU/assets/168362395/77248fad-234a-4288-a1b5-c70ff927e38e)
![OpenDTU Sysinfo](https://github.com/tbnobody/OpenDTU/assets/168362395/7cf6427d-7d2e-48bc-af82-ddf3f1f92520)

**Comment by ghost** [comment]
Da sich bisher keiner gemeldet hat, versuche ich mal etwas beizusteuern.

Fehler kommen immer nur am Tag, nachts ist der Wechselrichter nicht erreichbar.

Hat es überhaupt für längere Zeit funktioniert?
Wie sieht dein Hardwareaufbau aus, hast du den immer wiedermal erwähnten benötigten Kondensator von100µF in der Nähe des Funkmoduls verbaut?
Sind möglicherweise weitere Balkonkraftwerke und damit open DTUs in der Nähe? Hier könnte eine induviduelle Seriennummer deiner DTU helfen. Eigentlich sollten die zufällig sein, das klappt aber nicht.
Es ist auch eine  aktuellere Software v24.4.24 verfügbar.

**Reply by kerusker** [reply]
Vielen Dank für Deine Rückmeldung.  

Ja, die DTU läuft mehrere Tage am Stück und verliert dann plötzlich den Kontakt zum Wechselrichter und die Fehlermeldungen "Time Callibration" tauchen auf. 

Ich habe den Bausatz von blinkyparts und zwischen dem CMT Funkmodul und dem ESP32 sind insgesamt zwei Kondensatoren verlötet, 10 μF Kondensator & 100 nF Keramikkondensator. 

Ja, der Nachbar über mir hat exakt das gleiche Setup, jedoch mit DTU-Lite von Hoymiles und er hatte das Problem bisher nicht. 

Ich habe die Frequenz aktuell auf 868MhZ und vom Gefühl läuft die DTU stabiler. Ich habe bei Hoymile nachgefragt, auf welcher Frequenz die DTU-Lite meines Nachbarn sendet, um zu schauen, ob beide DTU sich hier durch die räumliche Nähe in die Quere kommen. Doch selbst der technische Support konnte diese Frage nicht beantworten, da in der UI der DTU-Lite keine Option besteht, die Frequenz manuell zum ändern. Der WR HM-1600 versteht 865-870MhZ. 

Die Seriennummer der DTU habe ich jetzt manuell verändert.

**Reply by Marc9485** [reply]
Könnte ne fehlerhafte Lötstelle zwischen ESP und CMT sein.
Anderes Netzteil(mit mehr Leistung) und USB-Kabel kannst de mal noch probieren. 
Hier hatte aber auch schon jemand über "falsche" Antennen berichtet die häufig bei CMT's dabei sind. Im inneren ist der Strahler bei diesen zu kurz. Anleitung um das zu beheben war auch mit in dem Post. 
Zeigt denn die DTU bei Syteminfo das CMT weiterhin als verbunden an? 

Bei Überlagerung der Signale der DTU's müsste dein Nachbar die gleichen Probleme haben(meiner Meinung nach... nicht, dass ich das weiß).

**Reply by kerusker** [reply]
Ich habe bereits alle Lötstellen einmal neu gelötet und gewissenhaft mit Lupe überprüft. 

Das Netzteil ist leistungsstark von einem Surface Tablett. 

Einen Konstruktionsfehler an der Antenne kann ich nicht ausschließen, doch wenn das System läuft, dann ist die Verbindung 100% stabil. Sie bricht dann plötzlich ab, ohne dass die DTU berührt wurde. Die Entfernung zum WR ist höchsten eine Spannweite, WR Richter hängt unter Modulen an der Außenfassade. 

Die CMT wird in der DTU-UI unverändert grün angezeigt, auch nach dem Kontaktverlust der DTU zum WR. 

Beim letzten Ausfall hatte ich das Gefühl, dass vorher das DC Netz für einen kurzen Augenschlag instabil war. Der Router in der selben Phase musste neu starten, die Uhren auf den anderen Phasen in der Wohnung (z.B. Backofen) hatten keinen Ausfall. Keine Sicherung ist heraus gefallen, es fühlt sich eher wie einen kurze Stromunterversorgung in der Phase des WR an. Der WR benötigt ja selbst die DC Spannung aus dem Netz, um zu laufen. Ein Spannungsfall im DC Netz bedeutet immer auch einen Einbruch der 50Hz Netzfrequenz. Möglicherweise nutzt der WR die stabil DC Netzfrequenz von 50Hz, um seine interne Zeitmessung zu kalibrieren? Kommt es zum kurzzeitigen Spannungsabfall, geht der time-sync zwischen WR und DTU verloren und wir haben einen "time calibration issue". Von der DTU können nach Kontaktabbruch auch keine Befehle mehr im WR platziert werden und so funktioniert auch die manuelle Zeit-Synchronisation im DTU-UI nicht. 

Ich Frage mich in diesem Fall, ob die Spannungsinstabilität vom Wechselrichter ausgelöst wurde, oder die Ursache Netzseitig liegt. Wir kennen diese Netz-Instabilität, wenn im Raum kurz die Deckenleuchten einmal flackern. Möglicherweise reagiert die Uhr im Backoden nicht so sensibel auf Spannungsschwankungen, wie der Wechselrichter?

**Reply by Sn0w3y** [reply]
Wenn dein Nachbar näher an deinem WR ist als du nicht :)

**Reply by Marc9485** [reply]
Du meinst AC-50Hz. DC ist Gleichspannung 😉
Wenn sich dein WR von den dir beschriebenen Netzschwankungen beeinträchtigen lässt dann müsste er eigentlich "Fehler Netzunterspannung und evtl. auch Netzunterfrequenz" anzeigen.🤔
Ich persönlich hatte das Problem mit dem NRF; Platine mit ESP, Elko, CMT(wird nicht gebraucht deshalb bei Hardware-Einstellungen nicht konfiguriert aber "besser haben als brauchen"😉),NRF, Display und LED's bestückt am orig. Apple 5W Netzteil. Da hatte ich dann jeden Tag mehrere Aussetzer. Dann hab ich direkt aufs NRF nen zusätzlichen Elko drauf gelötet, ins Gehäuse ne USB-C Buche mit 0,5mm Kabelquersch. eingebaut und ein 20Watt Netzteil mit Schnellladekabel daran angeschlossen. Seitdem hab ich ab und zu nur noch den Fehler. Hab mehrere Tage lang kein Problem und dann an einem WR 2-3 mal hintereinander und dann klappt alles wieder. Und 4 Tage später dann plötzlich an dem anderen WR...
Ob dir das hilt? KA🤷🏻‍♂️😅
![image](https://github.com/tbnobody/OpenDTU/assets/134625027/45332daa-b3a4-4d82-86c9-fc68130f3372)

**Comment by Sn0w3y** [comment]
Hast du ein Limit ?
Frägst du die API ab oder regelst du über die API?

**Reply by ghost** [reply]
Ich regle gar nicht, mein einzigster Wechselrichter läuft auf 100% durch, ich frage auch sonst nichts ab.

Was ich sagen kann ist, unterschiedliche Seriennummern haben mir geholfen,.
Ich setze 2 DTUs an dem WR ein. eine mit Display zum hinschauen und angeben, eine um an der DTU-Software herumzudoktern.
Ich konnte zeitweise mit der 2. DTU den Wechselrichter schlecht erreichen. Irgendwann habe ich festgestellt: für beide DTUs ist die selbe Seriennummer vergeben; nach dem Ändern einer klappt die Verbindung erheblich besser.

**Reply by tbnobody** [reply]
Siehe FAQ:
![image](https://github.com/tbnobody/OpenDTU/assets/1753222/8a31e3ff-6ff0-4bad-a76c-3d09e2ccb75b)

2 DTU's an einem WR können wohl funktionieren, es kann aber auch einfach nur Müll rauskommen.

**Reply by ghost** [reply]
Stimmt, steht da, habe ich vergessen zu erwähnen.
Liest keiner außer uns.

Ist aber nicht das Thema von @kerusker.
Mehrere DTUs, die verschiedene Wechselrichter bedienen, können sich auch gegenseitig behindern. Das warum ist 'was für dich @tbnobody. Damit ist der mittlere bis letzte Teil meiner Antwort so verkehrt nicht, siehe Diskussion [hier](https://github.com/tbnobody/OpenDTU/discussions/1953#discussioncomment-9270244).

Außerdem habe ich hier nur geantwortet, weil sich sonst keiner die letzen 8 Tage Gedanken zur Frage gemacht hat.

**Reply by kerusker** [reply]
Ja, mein WR ist auf 50% Limitiert. 
Ich frage alle 10Sekunden mit der DTU die Werte des WR ab.

**Reply by Sn0w3y** [reply]
Und genau DA liegt dein Problem..

Stoppe das einfach mal und du wirst sehen, dass das "Problem" kein OpenDTU Problem ist. Die "API" ist keine direkte "API" als solches. Diese wird nämlich auch vin der WebUI benutzt. 

Ich zitiere:

"The WebAPI is mainly generated to fetch and post information between the WebUI and the ESP. You use it at your own risk. It may change in future and break compatibility!"

**Reply by kerusker** [reply]
Sorry, aber ich habe gerade Schwierigkeiten deinem Gedanken inhaltlich zu folgen. Mein Issue ist "Time Calibration" und Kontaktverlust der DTU mit dem WR- 

Die OpenDTU fragt bei mir alle 10 Sekunden die Live Werte des WR ab, siehe DTU-Einstellungen/Abfrageintervall. Default liegt das Abfrageintervall auf 5s, ich habe es auf 10s erhöht. Diese Live-Darstellung der Wechselrichter-Werte ist doch das key-feature der DTU? Wenn ich diese "Stoppe", dann verliert die DTU ihr key-feature. Oder habe ich hier etwas grundlegend mißverstanden?

**Reply by ghost** [reply]
Nochmal ich,

Ob dir geholfen werden kann, weiß ich auch nicht.
Oftmals sind es banale Dinge, die die Kommunikation zw. DTU und WR schwierig machen.

[Kürzlich](https://github.com/tbnobody/OpenDTU/discussions/1964#discussioncomment-9337437) hat ein Austausch des Netzteils der DTU geholfen, die Komm. zu verbessern.

**Reply by Sn0w3y** [reply]
achso, dann hab ich das falsch verstanden. ich dachte machst das von extern.. dann vergiss das was ich geschrieben habe

**Comment by kerusker** [comment, marked as answer]
FEHLER SCHEINT GELÖST 

Nachdem ich folgende Anpassungen vorgenommen habe, trat der Fehler seit 2 Monaten NICHT mehr auf. Ich kann jedoch nicht sagen, was genau die Lösung des Problems war. 

Firmware: [v24.4.12] (https://github.com/tbnobody/OpenDTU/releases/tag/v24.4.12)

1. Ich habe die CMT2300A Frequenz von 865,00 MHz auf 868,00 MHz verändert - Tipp vom Hoymiles Support
2. Ich habe die default Seriennummer willkürlich verändert - Tipp aus einem Forum
3. Ich habe das Abfrage Intervall von default 5s auf 10s erhöht  - Tipp aus einem Forum

 Cheers, Carsten

**Reply by engels0n** [reply]
Hat bei mir auch geklappt.
Danke! :)

**Reply by Didi1965** [reply]
Bei mir hat Punkt 3. geholfen. Umstellung Polling von 2 auf 5 sec. Thanks! LG
